aws/sdk/examples/telephone-game/src/main.rs

```diff
@@ -0,0 +1,458 @@
+                    .await?;
+
+            info!("Transcription #{} == {}", i, &transcript);
+            current_phrase = transcript;
+
+            // Annotating the type here is necessary for using the '?' operator in this async block
+            // https://rust-lang.github.io/async-book/07_workarounds/02_err_in_async_blocks.html
+            Ok::<(), anyhow::Error>(())
+        })?;
+    }
+
+    // Log the final output
+    info!(
+        r#"The phrase
+"{}"
+became
+"{}"
+after {} iterations"#,
+        original_phrase, current_phrase, number_of_iterations
+    );
+
+    Ok(())
+}
+
+// Start a speech synthesis job and wait until it finishes before returning the URI of the audio file
+async fn synthesize_speech(
+    polly_client: &aws_sdk_polly::Client,
+    input_text: &str,
+    output_bucket_name: &str,
+) -> Result<String, anyhow::Error> {
+    let mut synthesis_task = polly_client
+        .start_speech_synthesis_task()
+        .text(input_text)
+        .voice_id(VoiceId::Joanna)
+        .output_format(OutputFormat::Mp3)
+        .output_s3_bucket_name(output_bucket_name)
+        .engine(Engine::Standard)
+        .send()
+        .await
+        .context("Failed to start speech synthesis task")?
+        .synthesis_task
+        .unwrap();
+
+    debug!(
+        "Waiting for speech synthesis task to complete. Timeout is {}s",
+        TASK_TIMEOUT_IN_SECONDS
+    );
+
+    let mut speech_synthesis_timeout_in_seconds = TASK_TIMEOUT_IN_SECONDS;
+
+    // Wait up to TASK_TIMEOUT_IN_SECONDS seconds for synthesis task to complete
+    // The status of the task is checked every TASK_WAIT_INTERVAL_IN_SECONDS in a loop
+    // Break out of the loop once the task succeeds or fails
+    'synthesis_task: loop {
+        speech_synthesis_timeout_in_seconds -= TASK_WAIT_INTERVAL_IN_SECONDS;
+        tokio::time::sleep(Duration::from_secs(TASK_WAIT_INTERVAL_IN_SECONDS as u64)).await;
+        synthesis_task = polly_client
+            .get_speech_synthesis_task()
+            .task_id(synthesis_task.task_id.unwrap())
+            .send()
+            .await
+            .context("Failed to check status of speech synthesis task")?
+            .synthesis_task
+            .unwrap();
+
+        use aws_sdk_polly::model::TaskStatus;
+        match synthesis_task.task_status.unwrap() {
+            TaskStatus::Completed => {
+                debug!("Speech synthesis task completed");
+                break 'synthesis_task;
+            }
+            TaskStatus::Failed => {
+                let reason = synthesis_task
+                    .task_status_reason
+                    .unwrap_or_else(|| "(no reason given)".to_owned());
+
+                bail!("Speech synthesis task failed with reason: {}", reason);
+            }
+            TaskStatus::InProgress | TaskStatus::Scheduled => {
+                debug!("Speech synthesis is ongoing...")
+            }
+            _ => unreachable!(),
```

Comment:
Do you think should we add a comment about why this branch is here? Also, would it be valuable to include the unknown status in the unreachable (for example, `unknown => unreachable!("{}", unknown)`)?